### ISSUE TITLE
In version 6.8, rollback does not work properly.

### ISSUE THREAD

**Issue by zeitakuhoshitsu** [deployphp/deployer#2524]
- Deployer version: v6.8.0
- Deployment OS:Amazon Linux release 2 (Karoo)
- Laravel 8

I'm sorry to keep posting, but I've posted the following issue.
https://github.com/deployphp/deployer/issues/2523

The reason I updated from 6.8 to 7 was because the rollback in 6.8 was not working properly.
I'm posting a new issue to tell you about the problem I found with the rollback in 6.8.

First, I did a deploy from the development environment to release 1 to 2 in the staging environment. Now, the staging site displays correctly.
Then when I did a rollback and checked the website, the site went blank and nothing was displayed.

I checked the staging directory.
When deploying, there were two directories: releases/1 and releases/2.

On rollback, there was only one, releases/1.

However, when I check .dep/releases, the line for 2 is still there, as shown below.

> 20210425093812,1
> 20210425093931,2

I think that line 2 should have been deleted, but isn't it?

When I tried to run php artisan view:clear in the staging environment, I got the following error message.

> PHP Warning: include(/var/www/staging.my-project/shared/vendor/composer/. /.. /... /releases/2/app/Console/Kernel.php): failed to open stream: No such file or directory in /var/www/staging.my-project/shared/vendor/composer/ ClassLoader.php on line 444

So now it looks like staging is still referring to the releases/2 directory, even though it did a rollback.

I did a rollback in debug mode, and the results are shown below.

```
[me@ip-x-x-x-x dev.my-progect]$ vendor/bin/dep rollback staging -vvv
➤ Executing task rollback
[local-staging] > cd /var/www/staging.my-progect && (if [ -d releases ] && [ "$(ls -A releases)" ]; then echo 'true'; fi)
[local-staging] < true
[local-staging] > cd /var/www/staging.my-progect && (cd releases && ls -t -1 -d */)
[local-staging] < 2/
[local-staging] < 1/
[local-staging] > cd /var/www/staging.my-progect && (if [ -f .dep/releases ]; then echo 'true'; fi)
[local-staging] < true
[local-staging] > cd /var/www/staging.my-progect && (tail -n 25 .dep/releases)
[local-staging] < 20210425093812,1
[local-staging] < 20210425093931,2
[local-staging] > cd /var/www/staging.my-progect && (if [[ $(man ln 2>&1 || ln -h 2>&1 || ln --help 2>&1) =~ '--relative' ]]; then echo 'true'; fi)
[local-staging] < true
[local-staging] > cd /var/www/staging.my-progect && (cd /var/www/staging.my-progect && ln -nfs --relative /var/www/staging.my-progect/releases/1 current)
[local-staging] > cd /var/www/staging.my-progect && (rm -rf /var/www/staging.my-progect/releases/2)
Rollback to `1` release was successful.
• done on [local-staging]
✔ Ok [184ms]
```

Can you tell me where the problem is?

**Comment by antonmedv** [comment]
> I think that line 2 should have been deleted, but isn't it?

No, this is correct. It's just a log of all deploys.

> When I tried to run php artisan view:clear in the staging environment, I got the following error message.

This is can be due to a `release` symlink pointing to old release, also to be run without call of `deploy:release` before. 

> Can you tell me where the problem is?

Everything looks okay. You see old content due to caching or opcode caching.

**Comment by zeitakuhoshitsu** [comment]
Thank you for your answer.

> No, this is correct. It's just a log of all deploys.

I now understand that the contents of .dep/releases is just a log.

> Everything looks okay. You see old content due to caching or opcode caching.

I tried clearing the OPcache, but it did not work correctly.

My application was still referring to the old directory, even though the symbolic link had been changed by the rollback command.

Then, after trying various things, I typed the following command in the staging environment, and it finally displayed correctly.

`$ composer dump-autoload`

I don't know why this made it appear correctly, and this may be a unique situation in my environment, but I am reporting it.
